src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php

```diff
@@ -35,4 +60,23 @@ public function getName()
     {
         return 'entity';
     }
+    
+    /**
+     * We consider two query builders with an equal SQL string and
+     * equal parameters to be equal.
+     * 
+     * @param QueryBuilder $queryBuilder
+     * 
+     * @return array
+     * 
+     * @internal This method is public to be usable as callback. It should not
+     *           be used in user code.
+     */
+    public function getQueryBuilderPartsForCachingHash($queryBuilder)
```

Comment:
please try moving both of them to protected. It should work, and anyway tests will tell if I'm wrong on 5.3 or any other version